python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_chaos_conditional_expectation.py

```diff
@@ -0,0 +1,668 @@
+#   reduced multi-index, the list of corresponding multi-indices from the equation :eq:`PCE_CE_3`.
+# - Aggregate the coefficients from the equation :eq:`PCE_CE_5`.
+# - The parametric PCE is defined by the equation :eq:`PCE_CE_4`.
+
+# %%
+# Conditional expectation
+# ~~~~~~~~~~~~~~~~~~~~~~~
+#
+# One method to reduce the input dimension of a function is to
+# consider its conditional expectation.
+# The conditional expectation function is:
+#
+# .. math::
+#
+#   \widehat{\model}(\physicalInputObservation_{\vect{u}})
+#   = \Expect{\model(\physicalInputObservation)
+#            \; | \; \physicalInputRandomVector_{\vect{u}}
+#            = \physicalInputObservation_{\vect{u}}}
+#
+# for any :math:`\physicalInputObservation_{\vect{u}} \in \Rset^{|\vect{u}|}`.
+# In general, there is no dedicated method to create such a conditional expectation
+# in the library.
+# We can, however, compute efficiently the conditional expectation of a polynomial
+# chaos expansion.
+# In turn, this conditional chaos expansion (PCE) is a polynomial chaos expansion
+# which can be computed using the :meth:`~openturns.FunctionalChaosResult.getConditionalExpectation`
+# method from the :class:`~openturns.FunctionalChaosResult` class.
+
+# %%
+# Create the PCE
+# ~~~~~~~~~~~~~~
+
+# %%
+import openturns as ot
+import openturns.viewer as otv
+from openturns.usecases import ishigami_function
+import matplotlib.pyplot as plt
+
+# %%
+# The next function creates a parametric PCE based on a
+# given PCE and a set of indices.
+
+
+# %%
+def MeanParametricPCE(chaosResult, indices):
+    """
+    Return the parametric PCE of Y with given input marginals set to the mean.
+
+    All marginal inputs, except those in the conditioning indices
+    are set to the mean of the input random vector.
+
+    The resulting function is :
+
+    g(xu) = PCE(xu, xnotu = E[Xnotu])
+
+    where xu is the input vector of conditioning indices,
+    xnotu is the input vector fixed indices and
+    E[Xnotu] is the expectation of the random vector of the components
+    not in u.
+
+    Parameters
+    ----------
+    chaosResult: ot.FunctionalChaosResult(inputDimension)
+        The polynomial chaos expansion.
+    indices: ot.Indices()
+        The indices of the input variables which are set to constant values.
+
+    Returns
+    -------
+    parametricPCEFunction : ot.ParametricFunction(reducedInputDimension, outputDimension)
+        The parametric PCE.
+        The reducedInputDimension is equal to inputDimension - indices.getSize().
+    """
+    inputDistribution = chaosResult.getDistribution()
+    if not inputDistribution.hasIndependentCopula():
+        raise ValueError(
+            "The input distribution has a copula" "which is not independent"
+        )
+    # Create the parametric function
+    pceFunction = chaosResult.getMetaModel()
+    xMean = inputDistribution.getMean()
+    referencePoint = [xMean[i] for i in indices]
```

Comment:
Thank you for pointing this. You are almost right, since there is the much simpler:
```python
import openturns as ot
pt = ot.Point(range(10))
indices = [5, 7, 9]
print(pt[indices])
```
In 9b11dd665, I took this into account.

Reply:
Neat!